### DISCUSSION TITLE
Speed benchmark

### DISCUSSION THREAD

**Discussion by xvjiarui** [arogozhnikov/einops#112] [Q&A] [answered]
This is an amazing project! Thank you!

Besides the beatify of einops, I also concern about the running time. 
Are there any speed benchmark results comparing with the original implementation?

**Comment by arogozhnikov** [comment, marked as answer]
@xvjiarui speed is similar, you can walk around different examples here:

Even on small examples speed is similar, and that's without tracing
https://github.com/arogozhnikov/einops/blob/master/docs/source_examples/Pytorch.ipynb

If you have really small tensors, just trace your model and you'll get optimal speed
